pontoon/tags/utils/tags.py

```diff
@@ -0,0 +1,94 @@
+
+from django.utils.functional import cached_property
+
+from pontoon.base.utils import glob_to_regex
+from pontoon.tags.models import Tag
+
+from .base import Clonable
+from .resources import TagsResourcesTool
+from .stats import TagsStatsTool
+from .tag import TagTool
+from .translations import TagsLatestTranslationsTool
+
+
+class TagsTool(Clonable):
+    """This provides an API for retrieving related ``Tags`` for given filters,
+    providing breakdowns of statistical information about the ``Tags``, and
+    managing ``Tags`` used in the site.
+    """
+
+    tag_class = TagTool
+    resources_class = TagsResourcesTool
+    translations_class = TagsLatestTranslationsTool
+    stats_class = TagsStatsTool
+    clone_kwargs = (
+        'locales',
+        'projects',
+        'priority',
+        'path',
+        'slug')
+
+    def __getitem__(self, k):
+        return self(slug=k)
+
+    def __iter__(self):
+        return self.iter_tags(self.stat_tool.data)
+
+    def __len__(self):
+        return len(self.stat_tool.data)
+
+    @property
+    def tag_manager(self):
+        return Tag.objects
```

Comment:
I see this pattern a lot in your code and I'm not sure I understand the purpose of it. Wouldn't it be easier to understand the rest of the code if we're explicit instead of using such "wrappers"?

Reply:
it makes the models extensible